src/Parser.php

```diff
@@ -48,7 +50,7 @@ private function parseIterable(iterable $lines): array
         return $patterns;
     }
 
-    private function parseLine(string $line): ?Pattern
+    private function parseLine(string $line, ?SourceInfo $sourceInfo): ?Pattern
```

Comment:
```suggestion
    private function parseLine(string $line, SourceInfo $sourceInfo): ?Pattern
```

This is only called from this class and with a `SourceInfo` object, so let's not make this nullable.